core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -105,6 +105,8 @@ private String metadataFileLocation(Table table) {
     if (table instanceof HasTableOperations) {
       TableOperations ops = ((HasTableOperations) table).operations();
       return ops.current().metadataFileLocation();
+    } else if (table instanceof BaseMetadataTable) {
+      return ((BaseMetadataTable) table).table().operations().current().metadataFileLocation();
```

Comment:
This is needed now since the metadata table won't enter above check of `HasTableOperations`